### PR TITLE
get current path via req.path instead of req.route.path

### PR DESCRIPTION
**Pull request by ptrwllrt**
fixing https://github.com/msiebuhr/node-statsd-client/issues/71

Seems like req.path will do the trick and return the currently matched path as expected, according to docs its supported in express 3/4.

### CODE REVIEW

**Comment by papandreou** [comment]
@cainvommars Nice work!

To prevent it from breaking in the future, could you please add a test with a server that uses routes defined this way? Looks like changing just one of the routes in the server set up in the `before` block of https://github.com/msiebuhr/node-statsd-client/blob/master/test/helpers.js could do the trick.

**Comment by cainvommars** [comment]
@papandreou sure thing - however, do tests run for you properly? I get a list or errors when running them locally. (node 6.10, npm 5)

`✖ 2 of 54 tests failed:

  1) EphemeralSocket Sends data immediately with maxBufferSize = 0:
     AssertionError: expected 8 to be close to 0 +/- 5
  

  2) Helpers response times GET /foo/:param/bar → "GET_foo_param_bar":
     Error: timeout of 2000ms exceeded
      at Timeout.<anonymous> (/Users/peter/git/tmp/node-statsd-client/node_modules/mocha/lib/runnable.js:158:14)
      at ontimeout (timers.js:380:14)
      at tryOnTimeout (timers.js:244:5)
      at Timer.listOnTimeout (timers.js:214:5)`

**Comment by papandreou** [comment]
The tests run fine for me. From the error messages it seems like you're running into some timing/timeout related problems. I've tried tweaking the timeouts on master -- could you try rebasing and see if that helps?

**Comment by cainvommars** [comment]
this should do the trick, 
what about all those js hint complaints btw?

**Comment by papandreou** [comment]
I don't see any jshint complaints? If they're about stuff in the node_modules folder, you might have run `jshint .`? I've added a .jshintignore on master now so that won't fail with hundreds of errors in the future :)

It could also be that this project doesn't have a .jshintrc, so in effect you're linting using a rule set located somewhere in your home directory. We should probably do something about that.

But since it builds fine on CI, I think this is good to merge in its current state!

**Comment by cainvommars** [comment]
yes, I also figured out that there are no rules right after posting the question ;-)

**Comment by msiebuhr** [comment]
Just a thought - we don't test against a route defined as a regex, which I imagine could be wired up differently internally in express.

**Comment by cainvommars** [comment]
added a testcase for regex defined routes, weird thing is i never get a positive test run for this one it('/foo/:param/bar → "GET_foo_param_bar"' it always a timeout,

**Comment by msiebuhr** [comment]
I have a hunch that `req.path` matches the materialized paths (i.e. what came in, not what the route is called inside express).

Could you also try cherry-picking https://github.com/msiebuhr/node-statsd-client/commit/f33029844980494fce8f5e812e3c4f3c4f71dc92 and see if that also fail?

Perhaps extending https://github.com/msiebuhr/node-statsd-client/blob/master/lib/helpers/getExpressMiddleware.js#L4 to accept arrays (or however else express does it internally):

    function sanitize(name) {
        var n = Array.isArray(name) ? name.map(sanitize).join("_") : name
        return n.replace(...)...;
   }

Or, if possible, pick just the matching path. That way it could semantically show up as a number of different paths.

**Comment by cainvommars** [comment]
I've updated the PR. The goal of this is to enable route definitions via array. We have three types of routes ("literal", "parameterised", "regex") all of them can occur in an array. 

- [x] support literal routes in array
- [ ] support parameterised routes in array (its not ideal yet: https://github.com/msiebuhr/node-statsd-client/pull/72/files#diff-9c9124fb66323f829b84dfd0bd727237R21)
- [x] support regex routes in array

Bonus:
- [ ] also added tests for regex routes

I expected the test for per regex defined routes to fail, have the same result locally as well. It would be great if one of you has a look at this :/ I've tried the new routes individually in a standalone express app and they all work and return 200s for me. Any idea what I could have missed?

File: test/FakeServer.js [hunk 1]
```diff
@@ -18,7 +19,7 @@ FakeServer.prototype.start = function (cb) {
     this._socket = dgram.createSocket('udp4');
 
     this._socket.on('message', function (msg, rinfo) {
-        //console.warn("Server got: '" + msg.toString() + "'");
+        debug("Server got: '" + msg.toString() + "'");
```
Comment (papandreou) [inline test/FakeServer.js]: Remember to add the debug module as a dev dep

Comment (msiebuhr) [inline test/FakeServer.js]: Huh? I'm using `require('util').debuglog('fakeserver')`. Behaves the same as the good 'ol `debug`-package.

**Pull request closed without merging**